gaj23/face-it
# Gaj/feature/error components

#### Purpose: 
> Check all that apply.

### What does this PR do? 
> Clearly and concisely describe changes made. 

This PR adds an Error component and provides conditional statements for when it should render. 

It closes Issue #10 #16 
It partially addresses issue #14

### Problems Encountered & Resulting Solutions  
> State any problems encountered 
> Describe your code changes in detail for reviewers.

We've discovered a bug in the code that we've decided not to address in this PR, but have created and issue for it. Please see below or issue #19 for further details.

### Where should the reviewer start?
> What file(s) and line(s) are where changes are made?

See App line 68 for conditional rendering

See Eco, Allergen Friendly, and Vegan lines 50 - 59 to see logic for rendering. 

The Error folder contains the component.

### How should this be tested?
> What steps need to be taken to confirm functionality? 

Please have the [local server](https://github.com/kelsiebesingeryeh/face-it-server) running and run `npm start` to click through the mentioned functionality. 
Then, comment out the fetch request to see the Error component render.

### Steps To Take
> Any questions and possible steps moving forward.

We discovered a bug: when a user tries to use the address bar to navigate into one of our components, the data does not persist. We have discussed how to solve this problem (localStorage? the React Hook of useEffect?) and have agreed to time box our work on it because we don't want it to get away from us. 


#### Checklist:
